src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,168 @@
+                // file logging
+                try {
+                    // maximum one 5MB file
+                    fileHandler = new FileHandler(logLocation, MAX_FILE_SIZE, FILE_COUNT);
+                    fileHandler.setFormatter(formatter);
+                    // note: fileHandler defaults to Level.ALL, so no need to set its log level
+                } catch (Exception e) {
+                    fileHandler = null;
+                    log.error("can't open the file handler:", e);
+                }
+            } else if (isStdErr(logLocation)) {
+                // console handler sending on stderr
+                // note that ConsoleHandler is sending on System.err
+                stderrHandler = new ConsoleHandler();
+                stderrHandler.setFormatter(formatter);
+                stderrHandler.setLevel(julLevel);
             }
         }
 
-        // replace default appender with the correct layout
-        LoggerConfig loggerConfig = config.getLoggerConfig(LogManager.ROOT_LOGGER_NAME);
-        loggerConfig.removeAppender(target);
+        // always have a console handler sending the logs to stdout
+        stdoutHandler = new StdoutConsoleHandler();
+        stdoutHandler.setFormatter(formatter);
+        stdoutHandler.setLevel(julLevel);
 
-        Appender appender = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(target)
-                    .withLayout(layout)
-                    .build();
-        appender.start();
+        // Create our Logger, and set our configured handlers on it
+        jmxfetchLogger = Logger.getLogger("org.datadog.jmxfetch");
+        jmxfetchLogger.setLevel(julLevel);
 
-        loggerConfig.addAppender(appender, null, null);
-        loggerConfig.setLevel(level);
+        if (fileHandler != null) {
+            jmxfetchLogger.addHandler(fileHandler);
+        }
+        if (stdoutHandler != null) { // always non-null but doesn't cost much
+            jmxfetchLogger.addHandler(stdoutHandler);
+        }
+        if (stderrHandler != null) {
+            jmxfetchLogger.addHandler(stderrHandler);
+        }
+    }
 
-        ctx.updateLoggers();
+    /** closeHandlers closes all opened handlers. */
+    public static synchronized void shutdown() {
+        for (Handler handler : jmxfetchLogger.getHandlers()) {
+            if (handler != null) {
+                handler.close();
+                jmxfetchLogger.removeHandler(handler);
+            }
+        }
     }
 
     /** Laconic logging for reduced verbosity. */
-    public static void laconic(org.slf4j.Logger logger, Level level, String message, int max) {
+    public static void laconic(org.slf4j.Logger logger, LogLevel level, String message, int max) {
         if (shouldLog(message, max)) {
-            if (level.isInRange(Level.ERROR, Level.ALL)) {
+            if (level == LogLevel.ERROR || level == LogLevel.ALL) {
```

Comment:
We should change this to:

```suggestion
            if (level == LogLevel.ERROR) {
```

There's no reason `laconic` would be called to log a message with a log level of `LogLevel.ALL`. Its only use today it 1 call with `LogLevel.WARN`.